pylint/testutils/output_line.py

```diff
@@ -15,16 +16,20 @@ class Message(
     def __new__(cls, msg_id, line=None, node=None, args=None, confidence=None):
         return tuple.__new__(cls, (msg_id, line, node, args, confidence))
 
-    def __eq__(self, other):
+    def __eq__(self, other: Union[object, "Message"]) -> bool:
         if isinstance(other, Message):
             if self.confidence and other.confidence:
                 return super().__eq__(other)
-            return self[:-1] == other[:-1]
+            return self[:-1] == other[:-1]  # type: ignore
```

Comment:
I could not get this to work... Perhaps somebody else wants to take a look at this issue and how to solve it!